napari/utils/io.py

```diff
@@ -290,13 +291,59 @@ def write_csv(
             writer.writerow(row)
 
 
+def csv_to_layer_data(filename: str) -> List[FullLayerData]:
+    """Return layer data from a csv.
+
+    Parameters
+    ----------
+    filename : str
+        Filename for reading csv.
+
+    Returns
+    -------
+    list of napari.types.FullLayerData or None
+        Layer full layer data or None.
+    """
+    # If extension is not csv then cannot read file
+    if not filename.endswith('.csv'):
+        return None
```

Comment:
```suggestion
```
since the only place this is called at the moment is `napari_get_reader`, where we have just checked this already.  I'm not sure it's necessary?  (but if we do keep it, should return an empty list instead of None)